models/User.php

```diff
@@ -352,6 +353,23 @@ public function attemptConfirmation($code)
     }
 
     /**
+     * Generates a new password and sends it to the user.
+     *
+     * @param string $code Confirmation code.
+     *
+     * @return boolean
+     */
+    public function resendPassword()
+    {
+        $generated_password = Password::generate(8);
```

Comment:
Agreed. I kept the second parameter of save to make sure no other attributes instead of password_hash can be touched in this operation.